tests/unit/selectMatrix.cpp

```diff
@@ -0,0 +1,393 @@
+/*
+ *   Copyright 2021 Huawei Technologies Co., Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License" );
+ * you may !use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <algorithm>
+#include <iostream>
+#include <numeric>
+#include <sstream>
+#include <vector>
+
+#include <graphblas.hpp>
+#include <utils/print_vec_mat.hpp>
+
+using namespace grb;
+
+#define STDERR_WITH_LINE std::cerr << "[Line " << __LINE__ << "]  "
+
+constexpr bool Debug = false;
+
+namespace {
+
+	template< bool enabled, typename D >
+	void printSparseMatrix(
+		const Matrix<D> &mat,
+		const std::string &name = "",
+		std::ostream &os = std::cerr
+	) {
+		if( !enabled ) return;
+		wait( mat );
+		print_matrix( mat, 256, name, os );
+	}
+
+} // namespace
+
+template< typename D >
+std::tuple< const size_t, const size_t, D > getMatrixEntry(
+	const std::pair< const std::pair<size_t, size_t >, D > &entry,
+	const typename std::enable_if<
+		!std::is_void< D >::value
+	>::type * = nullptr
+) {
+	return std::make_tuple(
+		entry.first.first,
+		entry.first.second,
+		entry.second
+	);
+}
+
+template< typename D >
+std::tuple< const size_t, const size_t, bool > getMatrixEntry(
+	const std::pair< const size_t, const size_t > &entry,
+	const typename std::enable_if< std::is_void< D >::value >::type * = nullptr
+) {
+	return std::make_tuple(
+		entry.first,
+		entry.second,
+		true
+	);
+}
+
+template<
+	typename D,
+	typename Func,
+	typename RIT, typename CIT,
+	Backend implementation
+>
+bool matrix_validate_predicate(
+	const Matrix< D, implementation, RIT, CIT > &B,
+	Func predicate,
+	typename std::enable_if< !std::is_void< D >::value >::type * = nullptr
+) {
+	/*
+	NOTE:
+	This function will fail for distributed backend because the local iterator of the matrix
+	does !reflect the global coordinates, which can lead to false negatives.
+	*/
```

Comment:
@byjtew to check, this comment is outdated right?

Reply:
Yes